synfig-studio/src/gui/widgets/widget_interpolation.cpp

```diff
@@ -42,34 +42,47 @@
 using namespace synfig;
 using namespace studio;
 
-Widget_Interpolation::Widget_Interpolation(Side side)
+//TODO(ice0): duplicated code
+static std::string
+interpolation_icon_name(synfig::Interpolation type)
 {
-	set_param_desc(
-		ParamDesc("interpolation")
-			.set_hint("enum")
-			.add_enum_value(INTERPOLATION_CLAMPED,"clamped",_("Clamped"))
-			.add_enum_value(INTERPOLATION_TCB,"auto",_("TCB"))
-			.add_enum_value(INTERPOLATION_CONSTANT,"constant",_("Constant"))
-			.add_enum_value(INTERPOLATION_HALT,"ease",
-							side == SIDE_BOTH? _("Ease In/Out") : (
-							side == SIDE_BEFORE? _("Ease In") :
-												 _("Ease Out")))
-			.add_enum_value(INTERPOLATION_LINEAR,"linear",_("Linear"))
-	);
-	set_icons();
+	switch(type)
+	{
+		case INTERPOLATION_CLAMPED:
+			return "interpolation_type_clamped_icon";
+		case INTERPOLATION_TCB:
+			return "interpolation_type_tcb_icon";
+		case INTERPOLATION_CONSTANT:
+			return "interpolation_type_const_icon";
+		case INTERPOLATION_HALT:
+			return "interpolation_type_ease_icon";
+		case INTERPOLATION_LINEAR:
+			return "interpolation_type_linear_icon";
+		case INTERPOLATION_MANUAL:
+		case INTERPOLATION_UNDEFINED:
+		case INTERPOLATION_NIL:
+		default:
+			break;
+	}
+	return "image-missing";
 }
 
-void Widget_Interpolation::set_icons()
+Widget_Interpolation::Widget_Interpolation(Side side)
 {
-	const unsigned int num_interpolations = 5;
-	// Must follow the same order of ParamDesc "interpolation" enum:
-	const char* interpolation_icons[num_interpolations] = {
-		"synfig-interpolation_type_clamped",
-		"synfig-interpolation_type_tcb",
-		"synfig-interpolation_type_const",
-		"synfig-interpolation_type_ease",
-		"synfig-interpolation_type_linear"
-	};
-	for (auto i = 0; i < num_interpolations; ++i)
-		set_icon(i, Gtk::Button().render_icon_pixbuf(Gtk::StockID(interpolation_icons[i]),Gtk::ICON_SIZE_MENU));
+	auto interpolation = ParamDesc("interpolation")
+		.set_hint("enum")
+		.add_enum_value(INTERPOLATION_CLAMPED,"clamped",_("Clamped"))
+		.add_enum_value(INTERPOLATION_TCB,"auto",_("TCB"))
+		.add_enum_value(INTERPOLATION_CONSTANT,"constant",_("Constant"))
+		.add_enum_value(INTERPOLATION_HALT,"ease",
+						side == SIDE_BOTH? _("Ease In/Out") : (
+								side == SIDE_BEFORE? _("Ease In") :
+								_("Ease Out")))
+		.add_enum_value(INTERPOLATION_LINEAR,"linear",_("Linear"));
```

Comment:
As you are rewriting it, let's add space after comma?

Reply:
Done. Please check. I also aligned them, but I can revert to one-space only.